Navbar padding, help panel (x), copy mults (issue 855,962)

juzen2003 opened the pull request:
- Fixes #962, #855 
- Were any Django, import pipeline, table_schema, or dictionary files modified? N
  - Database used: opus3_test_200107
  - All Django tests pass: Y
- Were any JavaScript or CSS files modified? Y
  - JSHINT run on all affected files: Y
  - Tested on Chrome, Firefox, Safari, and iOS: Y
    (Remember to test all browser sizes)
  - Tested for race conditions (with delayed API calls if applicable): NA

Description of changes:
- Fixed the issue to keep the spacing between Search/Browse Results/Cart/Detail the same even when the browser gets narrower.
- Fixed the issue that the position of "OPUS" (on the top left corner) will jump a bit when browser gets narrow.
- Make input texts and hints copiable in widgets with checkboxes and radio buttons.  
- Make sure "x" in the help panel is vertically aligned with the header text.

Known problems:
- Hold & copy texts and hints for checkboxes and radio buttons will not work in iOS . (It should work by default, haven't found the solution for this.)

juzen2003 committed:
Update the css styling to make sure the spacing between Search/Browse Results/
Cart/Detail in the navbar stays the same when the browser gets narrow.
- In opus.css (line 44-49), add the styling to keep the horizontal spacing
  between Search/Browse Results/Cart/Detail the same even when the browser
  size is narrow.
- In opus.css (line 71-76), add the styling to make sure the height of
  top navbar will not change when "Recent Announcement" appears & disappears.
- In opus.css (line 79-81), add the styling to make sure "OPUS" on the
  top left corner is not moving when browser gets narrow.

juzen2003 committed:
Make input texts and hints for checkboxes and radio buttons selectable
and copiable.
- In opus.css (line 2168-2175), add the styling to allow users to mouse
  over and select text in widgets.
- In widgets.js (line 54), add "cancel" options to specify the elements
  that will not trigger sort event when being grabbed.

juzen2003 committed:
Update comments in widgets.js (line 54-55)

juzen2003 committed:
Update the selector to be more specific for the stying to allow users to
mouse over and select text in widgets.
- In opus.css (line 2169), update the selector.

juzen2003 committed:
Update the selector (line 2169) in opus.css to avoid selecting the extra
space next to hints.

juzen2003 committed:
Add the styling to make sure "x" in help panel is vertically aligned with
the header text.
- In opus.css (line 1928-1931), add margin-bottom to align "x" and the
  header text in the help panel.

juzen2003 committed:
Update comments in opus.css (line 1928).

juzen2003 committed:
Center the title text and "x" icon in the help panel.
- In base.html (line 284), remove the padding-bottom styling. We would like
  to style the help panel header in opus.css.
- In opus.css (line 2015-2028), add styling to vertically center and align
  the title text and "x" icon in the help panel.

rfrenchseti reviewed (changes requested):
- The OPUS3 icon moves up and down slightly as the browser changes width.

juzen2003 committed:
Remove the display block styling for OPUS3 icon, after removing it, didn't
spot any OPUS3 jumping.

rfrenchseti reviewed (changes requested):
The size of the Search/Browse/Cart/Detail buttons has changed dramatically.

rfrenchseti commented on opus/application/static_media/js/widgets.js:
```diff
@@ -51,6 +51,9 @@ var o_widgets = {
         $("#op-search-widgets").sortable({
             items: "> li",
             cursor: "move",
+            // Specify the elements that will not trigger sorting when being grabbed. This is to make
```
Debby has a pull that prevents drag on anything except the widget title. I think that eliminates the need for this change.

rfrenchseti reviewed (changes requested):
Also, I know it's not one of these bugs, but while you're at it can you vertically center the (X) in the Select Metadata dialog?

juzen2003 committed:
Center the header text in the select metadata modal.
- In opus.css (line 685-688), add the styling to center the header text
  in the select metadata modal.
- In widgets.js (line 53), remove the cancel option in jquery ui sortable.

juzen2003 committed:
Restore the navlink button size for Search/Browse Results/Cart/Detail.
- In base.html (line 36), add op-blog-link class for the blog link anchor
  tag.
- In opus.css (line 72), remove the padding-top and bottom for blog link.
  This will make sure the height of top navbar will not change when
  "Recent Announcement" appears & disappears
